### ISSUE TITLE
Massive problem regarding steering wheel device index's

### ISSUE THREAD

**Issue by craytex** [JaydenMaalouf/JoystickPlugin#32]
When disconnecting a steering wheel and plugging it back in instead of it binding to "joystick0" It will keep "joystick0" but also add "joystick1" and it will only send axis and button events to the newly created joystick1, and again if i plug out and back in again it will make a "joystick2" and do the same thing again.

**Comment by craytex** [comment]
Hoping you see this, I would like to fix it myself but have no experience with sdl2

**Comment by JaydenMaalouf** [comment]
Hey,
Thanks for reporting the issue.
It seems there was some issues in the way that I was passing the data from the device mappings to the RemoveDevice method.
I've resolved the issues in a branch, see [here](https://github.com/JaydenMaalouf/JoystickPlugin/tree/fix/crashing-on-close)

I have created a pre-release with the fix, which you can download here: [release/2.0.1](https://github.com/JaydenMaalouf/JoystickPlugin/releases/tag/2.0.1-fix-crashing-on-close.1)

Please give this a go and let me know if it works as expected.
Thanks!

**Comment by craytex** [comment]
You are amazing for getting back to me so quickly, I really appreciate this
thank you so much! Will report back if it works

**Comment by craytex** [comment]
Strangely enough now its not detecting the steering wheel

**Comment by craytex** [comment]
Yep for some reason its not showing under connected devices under the joystick settings tab

**Comment by craytex** [comment]
Maybe when you fixed the joystick not getting removed when it should there was code that was calling it too early when being plugged in?

**Comment by craytex** [comment]
Went back and checked in the older version and device shows as expected

**Comment by craytex** [comment]
I am using unreal 5.0 btw

**Comment by craytex** [comment]
5.0.3

**Comment by JaydenMaalouf** [comment]
Hmmm. There is only change I made to the initialisation of the joysticks. I'll have to attempt to resolve it in the morning. I should have a fix up within the next 12hrs

**Comment by craytex** [comment]
Sounds good, thanks! Right now I can continue development on the older branch, Hoping its able to be fixed soon, Thank you

**Comment by JaydenMaalouf** [comment]
Hey @craytex,
The following version should resolve some issues with the joystick mappings: [release/2.0.1-fix-crashing-on-close.6](https://github.com/JaydenMaalouf/JoystickPlugin/releases/tag/2.0.1-fix-crashing-on-close.6)

**Comment by craytex** [comment]
Hey everything works except when you disconnect the wheel if you are running force feedback
Then it crashes with an error when closing again
Unhandled Exception: EXCEPTION_ACCESS_VIOLATION reading address 0x0000000000000050

SDL2
SDL2
UnrealEditor_JoystickPlugin!UForceFeedbackEffectBase::StopEffect() [D:\Agents\GitHub\_work\_temp\build\UE_5.0\HostProject\Plugins\JoystickPlugin\Source\JoystickPlugin\Private\ForceFeedback\Effects\ForceFeedbackEffectBase.cpp:194]
UnrealEditor_JoystickPlugin!UForceFeedbackEffectBase::DestroyEffect() [D:\Agents\GitHub\_work\_temp\build\UE_5.0\HostProject\Plugins\JoystickPlugin\Source\JoystickPlugin\Private\ForceFeedback\Effects\ForceFeedbackEffectBase.cpp:92]
UnrealEditor_JoystickPlugin!UForceFeedbackEffectBase::BeginDestroy() [D:\Agents\GitHub\_work\_temp\build\UE_5.0\HostProject\Plugins\JoystickPlugin\Source\JoystickPlugin\Private\ForceFeedback\Effects\ForceFeedbackEffectBase.cpp:26]
UnrealEditor_CoreUObject
UnrealEditor_CoreUObject
UnrealEditor_CoreUObject
UnrealEditor_CoreUObject
UnrealEditor_UnrealEd
UnrealEditor_UnrealEd
UnrealEditor_UnrealEd
UnrealEditor
UnrealEditor
UnrealEditor
UnrealEditor
UnrealEditor
UnrealEditor
kernel32
ntdll

Luckily this time the inputs all work

**Comment by JaydenMaalouf** [comment]
Hmm, seems I was attempting to stop the effect before destroying it. However, based on SDL documentation, this was unnecessary.
Please try the updated version here: [release/2.0.1-fix-crashing-on-close.11](https://github.com/JaydenMaalouf/JoystickPlugin/releases/tag/2.0.1-fix-crashing-on-close.11)

**Comment by craytex** [comment]
Apologies for the late reply,
This version gives out a new error message
Assertion failed: !IsUnreachable() [File:D:\build\++UE5\Sync\Engine\Source\Runtime\CoreUObject\Private\UObject\ScriptCore.cpp] [Line: 1839] WheelConstantSlip_C /Game/FGearAssets/Maps/UEDPIE_0_Multiplayer2.Multiplayer2:PersistentLevel.FGearPlayerController_C_0.JoystickMultiForceFeedback.WheelConstantSlip_C_0 Function: '/Script/JoystickPlugin.ForceFeedbackEffectBase:OnDestroyedEffect'

UnrealEditor_CoreUObject
UnrealEditor_JoystickPlugin!UForceFeedbackEffectBase::DestroyEffect() [D:\Agents\GitHub\_work\_temp\build\UE_5.0\HostProject\Plugins\JoystickPlugin\Source\JoystickPlugin\Private\ForceFeedback\Effects\ForceFeedbackEffectBase.cpp:117]
UnrealEditor_JoystickPlugin!UForceFeedbackEffectBase::BeginDestroy() [D:\Agents\GitHub\_work\_temp\build\UE_5.0\HostProject\Plugins\JoystickPlugin\Source\JoystickPlugin\Private\ForceFeedback\Effects\ForceFeedbackEffectBase.cpp:29]
UnrealEditor_CoreUObject
UnrealEditor_CoreUObject
UnrealEditor_CoreUObject
UnrealEditor_CoreUObject
UnrealEditor_UnrealEd
UnrealEditor_UnrealEd
UnrealEditor_UnrealEd
UnrealEditor
UnrealEditor
UnrealEditor
UnrealEditor
UnrealEditor
UnrealEditor
kernel32
ntdll

Im noticing its saying a wrong directory in these errors maybe thats a problem?

**Comment by JaydenMaalouf** [comment]
> Im noticing its saying a wrong directory in these errors maybe thats a problem?

Nah, the compile DLL doesn't know it's executing directory, so it uses the directory from the build agent.
I've uploaded a new version ([2.0.1-fix-crashing-on-close.21](https://github.com/JaydenMaalouf/JoystickPlugin/releases/tag/2.0.1-fix-crashing-on-close.21)) - please test and let me know how it goes!
I've had to add some compiler if statements to support multiple engine versions as it seems some of the safeguards were changed in UE5.0

**Comment by craytex** [comment]
Sorry for the long response time, This update seems to have fixed everything!
